django/db/models/fields/__init__.py

```diff
@@ -2072,7 +2081,19 @@ def formfield(self, **kwargs):
         return super(NullBooleanField, self).formfield(**defaults)
 
 
-class PositiveIntegerField(IntegerField):
+class PositiveIntegerRelDbTypeMixin(object):
+
+    def rel_db_type(self, connection):
+        """
+        Return related field database column data type.
+        """
+        if connection.features.related_fields_match_type:
+            return self.db_type(connection)
```

Comment:
I think it'd work if you put the mixin to the left of the base class where it's used.